addons/hr_work_entry_contract/wizard/hr_work_entry_regeneration_wizard.py

```diff
@@ -19,7 +19,8 @@ class HrWorkEntryRegenerationWizard(models.TransientModel):
     date_to = fields.Date('To', required=True, compute='_compute_date_to', store=True,
             readonly=False, default=lambda self: self.env.context.get('date_end'))
     employee_ids = fields.Many2many('hr.employee', string='Employees',
-                                    domain=lambda self: [('company_id', 'in', self.env.companies.ids)], required=True)
+            domain=lambda self: [('company_id', 'in', self.env.companies.ids), ('contract_id','!=', False)],
```

Comment:
contract_id could be false for employees who left the company. This domain would prevent regenerating work entries if you want to make a correction in the past.

Instead of restricting the values on the model definition, add a filter on the wizard, and select it by default.